core/src/main/java/org/infinispan/security/impl/ClusterRoleMapper.java

```diff
@@ -34,29 +49,29 @@
    @Override
    public void setContext(PrincipalRoleMapperContext context) {
       this.cacheManager = context.getCacheManager();
-      clusterRegistry = cacheManager.getGlobalComponentRegistry().getComponent(ClusterRegistry.class);
+      GlobalConfiguration globalConfiguration = cacheManager.getGlobalComponentRegistry().getGlobalConfiguration();
+      CacheMode cacheMode = globalConfiguration.isClustered() ? CacheMode.REPL_SYNC : CacheMode.LOCAL;
+      ConfigurationBuilder cfg = new ConfigurationBuilder();
```

Comment:
The ClusterRegistry also disabled authorization for internal caches, which no longer happens here. Was that an unwanted side effect?